README.md

```diff
@@ -5,6 +5,65 @@ This repository is aimed to be added as git submodule into particular
 containers' source repositories.  By default, the path to submodule should be
 named 'common'.
 
+Usage
+-----
+
+This section explains the usage of the shared scripts in this repository when
+it is used as a submodule in a container source repository.
+
+Once you have the repository set as a submodule include `common.mk` into your
+root Makefile in order to access the default rules used to call shared scripts.
+
+**Default rules:**
+
+`make` or `make build`  
+This rule will build an image without tagging it with any tags after it is built.
+`make build` will also expect a `README.md` so that it can transfrom it into
+a man page that gets added to the image so make sure it is available and that
+you have the `go-md2man` tool installed on your host.
+
+`make tag`  
+Use this rule if you want to tag an image after it is built. It will be tagged with
+two tags - name:latest and name:version.
+Depends on `build`
+
+`make test` or `make check`  
+This rule will run the testsuite scripts contained in the container source repositories.
+It expects the test to be available at `$gitroot/$version/test/run`
+Depends on `tag` as some tests might need to have the images tagged (s2i).
```

Comment:
I pretty much dislike this api ... I would much rather allow users to not have the $root/$version/test directory, and allow them to use only $root/test(s) dir.., wdyt?

Reply:
No strong opinion about this. We will still need to have the tests inside the git downstream but I can work around this in the tooling.